src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,526 @@
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setUsername(String username) {
+        this.username = username;
+        return this;
+    }
+
+    /**
+     * Sets the password for the client connection. The password is the one for the username or
+     * API key set by the {@link #setUsername(String)} method.
+     *
+     * @param password user password or API key passphrase
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setPassword(String password) {
+        this.password = password;
+        return this;
+    }
+
+    /**
+     * Set a custom GsonBuilder to use when serializing and de-serializing requests and
+     * responses between the CloudantClient and the server.
+     * <P>
+     * Note: the supplied GsonBuilder will be augmented with some internal TypeAdapters.
+     * </P>
+     *
+     * @param gsonBuilder the custom GsonBuilder to use
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setGsonBuilder(GsonBuilder gsonBuilder) {
+        this.gsonBuilder = gsonBuilder;
+        return this;
+    }
+
+    /**
+     * Set the maximum number of connections to maintain in the connection pool.
+     * <P>
+     * Note: this setting only applies if using the optional OkHttp dependency. If OkHttp is not
+     * present then the JVM configuration is used for pooling. Consult the JVM documentation for
+     * the {@code http.maxConnections} property for further details.
+     * </P>
+     * Defaults to {@link #DEFAULT_MAX_CONNECTIONS}
+     *
+     * @param maxConnections the maximum number of simultaneous connections to open to the server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setMaxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+        return this;
+    }
+
+    /**
+     * Sets a proxy url for the client connection.
+     *
+     * @param proxyURL the URL of the proxy server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setProxyURL(URL proxyURL) {
+        this.proxyURL = proxyURL;
+        return this;
+    }
+
+    /**
+     * Sets an optional proxy username for the client connection.
+     *
+     * @param proxyUser username for the proxy server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setProxyUser(String proxyUser) {
+        this.proxyUser = proxyUser;
+        return this;
+    }
+
+    /**
+     * Sets an optional proxy password for the proxy user specified by
+     * {@link #setProxyUser(String)}.
+     *
+     * @param proxyPassword password for the proxy server user
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setProxyPassword(String proxyPassword) {
```

Comment:
I don't mind as long as they are consistent, but it appears that people generally prefer not having the `set` prefix.